firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/PersonalizationTest.java

```diff
@@ -0,0 +1,213 @@
+              .withPersonalizationMetadata(
+                  new JSONObject(
+                      "{key1: {personalizationId: 'id1'}, key2: {personalizationId: 'id2'}}"))
+              .build();
+    } catch (JSONException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static final List<Bundle> FAKE_LOGS = new ArrayList<>();
+
+  private Personalization personalization;
+
+  @Mock private AnalyticsConnector mockAnalyticsConnector;
+  @Mock private FirebaseInstallationsApi mockFirebaseInstallations;
+  @Mock private ConfigCacheClient mockFetchedCache;
+  @Mock private ConfigCacheClient mockActivatedCache;
+  @Mock private ConfigCacheClient mockDefaultsCache;
+  @Mock private ConfigFetchHandler mockFetchHandler;
+  @Mock private ConfigMetadataClient metadataClient;
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+
+    doAnswer(invocation -> FAKE_LOGS.add(invocation.getArgument(2)))
+        .when(mockAnalyticsConnector)
+        .logEvent(
+            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
+
+    personalization = new Personalization(mockAnalyticsConnector);
+
+    FAKE_LOGS.clear();
+  }
+
+  @Test
+  public void logArmActive_nonPersonalizationKey_notLogged() {
+    personalization.logArmActive("key3", CONFIG_CONTAINER);
+
+    verify(mockAnalyticsConnector, times(0))
+        .logEvent(
+            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
+    assertThat(FAKE_LOGS).isEmpty();
+  }
+
+  @Test
+  public void logArmActive_singlePersonalizationKey_loggedOnce() {
+    personalization.logArmActive("key1", CONFIG_CONTAINER);
+
+    verify(mockAnalyticsConnector, times(1))
+        .logEvent(
+            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
+    assertThat(FAKE_LOGS).hasSize(1);
+
+    Bundle params = new Bundle();
+    params.putString(ARM_KEY, "id1");
+    params.putString(ARM_VALUE, "value1");
+    assertThat(FAKE_LOGS.get(0).toString()).isEqualTo(params.toString());
+  }
+
+  @Test
+  public void logArmActive_multiplePersonalizationKeys_loggedMultiple() {
+    personalization.logArmActive("key1", CONFIG_CONTAINER);
+    personalization.logArmActive("key2", CONFIG_CONTAINER);
+
+    verify(mockAnalyticsConnector, times(2))
+        .logEvent(
+            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
+    assertThat(FAKE_LOGS).hasSize(2);
+
+    Bundle params1 = new Bundle();
+    params1.putString(ARM_KEY, "id1");
+    params1.putString(ARM_VALUE, "value1");
+    assertThat(FAKE_LOGS.get(0).toString()).isEqualTo(params1.toString());
+
+    Bundle params2 = new Bundle();
+    params2.putString(ARM_KEY, "id2");
+    params2.putString(ARM_VALUE, "value2");
+    assertThat(FAKE_LOGS.get(1).toString()).isEqualTo(params2.toString());
+  }
+
+  @Test
+  public void logArmActive_RemoteConfigIntegration_successful() {
```

Comment:
Maybe this should go with the FirebaseRemoteConfig tests since it's testing the functionality of the `get` methods? Then the class wouldn't need `@VisibleForTesting`.

Reply:
Done.